src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -859,8 +876,36 @@ def update_agents(cmd, client, resource_group_name, cluster_name, https_proxy=""
     # Setting the config dataplane endpoint
     config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
 
+    if auto_upgrade != "false":
+        helm_values = get_all_helm_values(release_namespace, kube_config, kube_context)
+        auto_upgrade = helm_values.get('systemDefaultValues').get('azureArcAgents').get('autoUpdate')
+        if not auto_upgrade:
+            auto_upgrade = "false"
+
+    if default_registry:
+        container_registry_repository = None
+    else:
+        isCustomRegistryEnabled = helm_values.get('global').get('isCustomRegistryEnabled')
+        if isCustomRegistryEnabled is True and container_registry_repository is None:
+            container_registry_repository = helm_values.get('systemDefaultValues').get('image').get('repository')
+        if anonymous_pull is not None:
```

Comment:
This case should also be part of initial parameter validations. We should throw error if both anonymous_pull and container_registry_username/container_registry_password are set

Reply:
will add parameter validation